test/funnels/automation/StableSwapper.t.sol

```diff
@@ -0,0 +1,129 @@
+               count: 10,
+                 lot: uint112(10_000 * WAD), 
+              reqOut: uint112(9900 * 10**6)
+        }));
+        stableSwapper.setConfig(USDC, DAI, StableSwapper.PairConfig({ 
+               count: 10,
+                 lot: uint96(10_000 * 10**6), 
+              reqOut: uint112(9900 * WAD)
+        }));
+        stableSwapper.permit(KEEPER);
+        vm.stopPrank();
+    }
+
+    function testConstructor() public {
+        StableSwapper s = new StableSwapper(address(0xABC));
+        assertEq(address(s.swapper()),  address(0xABC));
+        assertEq(s.wards(address(this)), 1);
+    }
+
+    function testAuth() public {
+        checkAuth(address(stableSwapper), "StableSwapper");
+    }
+
+    function testSwapByKeeper() public {
+        vm.warp(block.timestamp + 3600);
+        uint256 prevDst = GemLike(DAI).balanceOf(address(buffer));
+
+        vm.expectEmit(true, true, true, false);
+        emit Swap(address(stableSwapper), USDC, DAI, 0, 0);
+        vm.prank(KEEPER); uint256 out = stableSwapper.swap(USDC, DAI, 9900 * WAD, address(uniV3Callee), USDC_DAI_PATH);
+
+        assertGe(out, 9900 * WAD);
+        assertEq(GemLike(DAI).balanceOf(address(buffer)), prevDst + out);
+        assertEq(GemLike(DAI).balanceOf(address(stableSwapper)), 0);
+        assertEq(GemLike(USDC).balanceOf(address(stableSwapper)), 0);
+        assertEq(GemLike(DAI).balanceOf(address(swapper)), 0);
+        assertEq(GemLike(USDC).balanceOf(address(swapper)), 0);
+        assertEq(GemLike(DAI).balanceOf(address(uniV3Callee)), 0);
+        assertEq(GemLike(USDC).balanceOf(address(uniV3Callee)), 0);
+
+        vm.warp(block.timestamp + 3600);
+        prevDst = GemLike(USDC).balanceOf(address(buffer));
+
+        vm.expectEmit(true, true, true, false);
+        emit Swap(address(stableSwapper), DAI, USDC, 0, 0);
+        vm.prank(KEEPER); out = stableSwapper.swap(DAI, USDC, 9900 * 10**6, address(uniV3Callee), DAI_USDC_PATH);
+
+        assertGe(out, 9900 * 10**6);
+        assertEq(GemLike(USDC).balanceOf(address(buffer)), prevDst + out);
+        assertEq(GemLike(DAI).balanceOf(address(stableSwapper)), 0);
+        assertEq(GemLike(USDC).balanceOf(address(stableSwapper)), 0);
+        assertEq(GemLike(DAI).balanceOf(address(swapper)), 0);
+        assertEq(GemLike(USDC).balanceOf(address(swapper)), 0);
+        assertEq(GemLike(DAI).balanceOf(address(uniV3Callee)), 0);
+        assertEq(GemLike(USDC).balanceOf(address(uniV3Callee)), 0);
+    }
+
+    function testSwapExceedingCount() public {
+        vm.expectRevert("StableSwapper/exceeds-count");
+        vm.prank(KEEPER); stableSwapper.swap(USDC, USDC, 0, address(uniV3Callee), USDC_DAI_PATH);
+    }
+
+    function testSwapWithMinTooSmall() public {
+        (,,uint256 reqOut) = stableSwapper.configs(USDC, DAI);
+        vm.expectRevert("StableSwapper/min-too-small");
+        vm.prank(KEEPER); stableSwapper.swap(USDC, DAI, reqOut - 1, address(uniV3Callee), USDC_DAI_PATH);
+    }
+}
```

Comment:
Some missing tests/checks:

- we should explicitly check permit and forbid (and their events)
- we should explicitly check kiss and diss (and their events)
- we should explicitly check setConfig and its event
- we should somehow check that count is decreasing after a swap
- we should check the case where we pass minOut as 0
- we should check the case where an un-allowed actor calls setConfig

Reply:
Implemented at https://github.com/makerdao/dss-allocator/pull/14.